### PR TITLE
Shorten contact tests and add friction to skipped tests

### PR DESCRIPTION
**Pull request by recuero**
#22532

### CODE REVIEW

File: modules/contact/test/tests/mortar_tm/2d/ad_frictional/finite.i [hunk 1]
```diff
@@ -152,9 +152,9 @@ name = 'finite'
   petsc_options = '-snes_converged_reason -ksp_converged_reason'
   petsc_options_iname = '-pc_type -pc_factor_shift_type -pc_factor_shift_amount'
   petsc_options_value = 'lu        NONZERO               1e-15'
-  end_time = 13.5
-  dt = 0.1
-  dtmin = 0.1
+  end_time = 5.3 # 13.5
```
Comment (lindsayad) [inline modules/contact/test/tests/mortar_tm/2d/ad_frictional/finite.i:155]:
```suggestion
  end_time = 5.3
```

File: modules/contact/test/tests/mortar_tm/2d/ad_frictional/finite_rr.i [hunk 2]
```diff
@@ -159,9 +159,9 @@ name = 'finite_rr'
   petsc_options = '-snes_converged_reason -ksp_converged_reason'
   petsc_options_iname = '-pc_type -pc_factor_shift_type -pc_factor_shift_amount'
   petsc_options_value = 'lu        NONZERO               1e-15'
-  end_time = 13.5
-  dt = 0.1
-  dtmin = 0.1
+  end_time = 5.3 # 13.5
```
Comment (lindsayad) [inline modules/contact/test/tests/mortar_tm/2d/ad_frictional/finite_rr.i:162]:
```suggestion
  end_time = 5.3
```

File: modules/contact/test/tests/mortar_tm/2d/ad_frictional/finite_stiff.i [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+  []
+[]
+
+[Executioner]
+  type = Transient
+  solve_type = 'NEWTON'
+  petsc_options = '-snes_converged_reason -ksp_converged_reason'
+  petsc_options_iname = '-pc_type -pc_factor_shift_type -pc_factor_shift_amount'
+  petsc_options_value = 'lu        NONZERO               1e-24'
+  end_time = 5.3 # 13.5
```
Comment (lindsayad) [inline modules/contact/test/tests/mortar_tm/2d/ad_frictional/finite_stiff.i:156]:
```suggestion
  end_time = 5.3
```

File: modules/contact/test/tests/mortar_tm/2d/ad_frictional/finite_stiff.i [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+  dt = 0.12
+  dtmin = 0.12
+  line_search = 'none'
+  nl_div_tol = 1e+100
```
Comment (lindsayad) [inline modules/contact/test/tests/mortar_tm/2d/ad_frictional/finite_stiff.i:160]:
```suggestion
  nl_div_tol = 1e100
```
Was this actually necessary? I forget what the default is

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: modules/contact/test/tests/mortar_tm/2d/ad_frictional/finite_stiff.i (hunk 4 above)
Comment (recuero) [reply modules/contact/test/tests/mortar_tm/2d/ad_frictional/finite_stiff.i]: When I converted the tests, I removed this option only to realize it was necessary. Atom autocomplete tells me the default value is `1e10`. It's relatively easy to go beyond that with contact constraints.

**Comment by recuero** [comment]
@tophmatthews Shortened these tests to bounce 3/4 times, i.e. end time is 5.3 instead of 13.5. I believe that should be enough to prevent capability regression. As you pointed out, the frictional tests are tricky to set up. I suspect the time steps (still) used here are very large to assume the initial guess is close to the actual solution. So, in practice, time step cuts are not unexpected.

**Review by lindsayad — APPROVED** [review]

**Comment by tophmatthews** [comment]
> end time is 5.3 instead of 13.5

Hate to say it, but you will still miss some difficult bounces that occur after this, something with the mesh alignment that caused problems in the past. I'll leave it to you and @lindsayad to either live with that, or perhaps split the test into multiple shorter tests where the block starts at different places, and only bounces once.

**Comment by recuero** [comment]
If the issues are with mesh alignment, those should be covered by other tests in similar folders that run until the original end time. I shortened the frictional tests that, understandably, were taking longer and timing out in pull requests.

**Comment by lindsayad** [comment]
I am fine with the holistic coverage we have. @recuero feel free to merge when ready

**Pull request merged**